src/main/kotlin/build/buf/intellij/manifest/Manifest.kt

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.intellij.manifest
+
+import com.intellij.openapi.vfs.VirtualFile
+import java.util.*
+import kotlin.collections.ArrayList
+import kotlin.collections.HashMap
+import kotlin.collections.HashSet
+
+class Manifest(
+    /* Contains digest hex to list of relative paths */
+    private val digestToPaths: Map<String, List<String>>,
+    /* Contains relative path to Digest */
+    private val pathToDigest: Map<String, Digest>,
+    /* Contains mapping of path to canonical path */
+    private val pathToCanonicalPath: Map<String, String>
+) {
+    /**
+     * Returns all unique paths in the manifest, order not guaranteed.
+     */
+    fun getPaths(): Set<String> {
+        return Collections.unmodifiableSet(pathToDigest.keys)
```

Comment:
`setOf()` should do a similar thing if you're interested

Reply:
Took a look at this and there is a https://github.com/JetBrains/kotlin/blob/924c28507067cbfbf78a6509ea89eabe496e34ca/libraries/stdlib/src/kotlin/collections/Sets.kt#L38-L44 which defers to https://github.com/JetBrains/kotlin/blob/ec1553ae4123974f50f4f18a1b6f2656cac2dde1/libraries/stdlib/common/src/generated/_Arrays.kt#L10119-L10130, however that would make a copy each time it is called. This way it will just prevent mutation of the underlying map.